### PR TITLE
feat(Guild/Commands): guild rename

### PR DESCRIPTION
**Pull request by Kitzunu**
* cherry-pick commit (https://github.com/TrinityCore/TrinityCore/commit/678cade1163080263bdea9e349fbb67b7af15ea7#)

## Changes Proposed:
-  .guild rename "oldName" "newName"

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/10319

## Tests Performed:

- tested in-game

## How to Test the Changes:

1. .guild rename "oldName" "newName"
2. Restart your client

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Bogir — CHANGES_REQUESTED** [review]

File: src/server/scripts/Commands/cs_guild.cpp [hunk 1]
```diff
@@ -206,6 +207,55 @@ class guild_commandscript : public CommandScript
         return targetGuild->ChangeMemberRank(player->GetGUID(), rank);
     }
 
+    static bool HandleGuildRenameCommand(ChatHandler* handler, char const* _args)
```
Comment (Winfidonarleyan) [inline src/server/scripts/Commands/cs_guild.cpp:210]: no no no, please use new cc api

File: src/server/game/Guilds/Guild.h [hunk 2]
```diff
@@ -686,6 +686,8 @@ class Guild
     std::string const& GetMOTD() const { return m_motd; }
     std::string const& GetInfo() const { return m_info; }
 
+    bool SetName(std::string const& name);
```
Comment (Winfidonarleyan) [inline src/server/game/Guilds/Guild.h:689]: std::string_view

**Comment by dualcorex** [comment]
I tested this in-game three times. Confirmed that it works. 

NOTE: Guild name does NOT update automatically, however it will update by logging to creator screen and right back in. Client does NOT have to be restarted for guild name to update.

.guild rename "Name1" "Name2", log to creator screen, name changed to the new name
.guild rename "Name2" "Name1", log to creator screen, name changed back to the original name

File: src/server/scripts/Commands/cs_guild.cpp (hunk 1 above)
Comment (Kitzunu) [reply src/server/scripts/Commands/cs_guild.cpp:210]: I will not convert the whole file for this.

**Comment by Bogir** [comment]
Merging without autoupdate? Nice work!
ok

**Comment by Nyeriah** [comment]
> Merging without autoupdate? Nice work! ok

I'm sorry, I don't get it. What do you mean exactly?

File: src/server/game/Guilds/Guild.cpp [hunk 3]
```diff
@@ -1173,6 +1173,21 @@ void Guild::OnPlayerStatusChange(Player* player, uint32 flag, bool state)
     }
 }
 
+bool Guild::SetName(std::string const& name)
```
Comment (Nyeriah) [inline src/server/game/Guilds/Guild.cpp:1176]:
```suggestion
bool Guild::SetName(std::string_view const& name)
```

File: src/server/scripts/Commands/cs_guild.cpp (hunk 1 above)
Comment (Nyeriah) [reply src/server/scripts/Commands/cs_guild.cpp:210]: It's fine

**Comment by Bogir** [comment]
> > Merging without autoupdate? Nice work! ok
> 
> I'm sorry, I don't get it. What do you mean exactly?

if i had a sniffer I'd say what packet you ignore. Packet which broadcast changes in Guild Name. When character create guild or when you meet other character, or when he is entering world - it's not need to restart server to watch name of his Guild

**Comment by Kitzunu** [comment]
No, you just need to relog. Probably due to session cache.

For the PR in it self, it does what it is supposed to. I will personally not dig into packets as I know nothing about them, but if someone wants to do that, I think it would be cool

**Comment by Nyeriah** [comment]
> > > Merging without autoupdate? Nice work! ok
> > 
> > 
> > I'm sorry, I don't get it. What do you mean exactly?
> 
> if i had a sniffer I'd say what packet you ignore. Packet which broadcast changes in Guild Name. When character create guild or when you meet other character, or when he is entering world - it's not need to restart server to watch name of his Guild

Hey,

Thanks for your concern, but it seems this PR doesn't require a server restart. It does, however, require you to relog or rezone in order for the guild name to update. So the packets you mention are being received normally, there's no issues with them

**Review by Nyeriah — APPROVED** [review]

**Comment by Bogir** [comment]
> > > > Merging without autoupdate? Nice work! ok
> > > 
> > > 
> > > I'm sorry, I don't get it. What do you mean exactly?
> > 
> > 
> > if i had a sniffer I'd say what packet you ignore. Packet which broadcast changes in Guild Name. When character create guild or when you meet other character, or when he is entering world - it's not need to restart server to watch name of his Guild
> 
> Hey,
> 
> Thanks for your concern, but it seems this PR doesn't require a server restart. It does, however, require you to relog or rezone in order for the guild name to update. So the packets you mention are being received normally, there's no issue with them

GuildName updates only if character make some actions (zone changing and so on) but when you create guild - guild name instantly appears above your head. Your character must not go somewhere to see changes. Can you find this packet? Or you won't?

**Comment by 55Honey** [comment]
> > > > > Merging without autoupdate? Nice work! ok
> > > > 
> > > > 
> > > > I'm sorry, I don't get it. What do you mean exactly?
> > > 
> > > 
> > > if i had a sniffer I'd say what packet you ignore. Packet which broadcast changes in Guild Name. When character create guild or when you meet other character, or when he is entering world - it's not need to restart server to watch name of his Guild
> > 
> > 
> > Hey,
> > Thanks for your concern, but it seems this PR doesn't require a server restart. It does, however, require you to relog or rezone in order for the guild name to update. So the packets you mention are being received normally, there's no issue with them
> 
> GuildName updates only if character make some actions (zone changing and so on) but when you create guild - guild name instantly appears above your head. Your character must not go somewhere to see changes. Can you find this packet? Or you won't?
> 
> 2 moderators (GMs) watch guildName "F**cik server" and whant to fix this "BadName" Moderator1 fix it first, moderator2 will receive error. He don't know what "BadName" allready changed

It's the exact same issue with character renames. Maybe create a new issue about it?

File: src/server/game/Guilds/Guild.h [hunk 4]
```diff
@@ -685,6 +685,8 @@ class Guild
     std::string const& GetMOTD() const { return m_motd; }
     std::string const& GetInfo() const { return m_info; }
 
+    bool SetName(std::string_view const& name);
```
Comment (Winfidonarleyan) [inline src/server/game/Guilds/Guild.h:688]: need `std::string_view name`

**Pull request merged**